src/plugins/ietf/ietf_interface.cpp

```diff
@@ -0,0 +1,530 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string>
+#include <exception>
+#include <memory>
+
+#include <vom/interface.hpp>
+#include <vom/om.hpp>
+#include <vom/l3_binding.hpp>
+#include <vom/route.hpp>
+
+#include <vpp-oper/interface.hpp>
+
+#include "sc_plugins.h"
+
+using VOM::interface;
+using VOM::OM;
+using VOM::HW;
+using VOM::l3_binding;
+using VOM::rc_t;
+
+/**
+ * @brief Callback to be called by any config change of
+ * "/ietf-interfaces:interfaces/interface/enabled" leaf.
+ */
+static int
+ietf_interface_enable_disable_cb(sr_session_ctx_t *session, const char *xpath,
+                                 sr_notif_event_t event, void *private_ctx)
+{
+    UNUSED(private_ctx);
+    char *if_name = nullptr;
+    sr_change_iter_t *iter = nullptr;
+    sr_change_oper_t op = SR_OP_CREATED;
+    sr_val_t *old_val = nullptr;
+    sr_val_t *new_val = nullptr;
+    sr_xpath_ctx_t xpath_ctx = { 0, };
+    int rc = SR_ERR_OK, op_rc = SR_ERR_OK;
+    std::shared_ptr<interface> intf;
+
+    SRP_LOG_INF("In %s", __FUNCTION__);
+
+    /* no-op for apply, we only care about SR_EV_{ENABLED,VERIFY,ABORT} */
+    if (SR_EV_APPLY == event)
+        return SR_ERR_OK;
+
+    SRP_LOG_DBG("'%s' modified, event=%d", xpath, event);
+
+    /* get changes iterator */
+    rc = sr_get_changes_iter(session, xpath, &iter);
+    if (SR_ERR_OK != rc) {
+        sr_free_change_iter(iter);
+        SRP_LOG_ERR("Unable to retrieve change iterator: %s", sr_strerror(rc));
+        return rc;
+    }
+
+    foreach_change (session, iter, op, old_val, new_val) {
+
+        SRP_LOG_DBG("A change detected in '%s', op=%d",
+                    new_val ? new_val->xpath : old_val->xpath, op);
+        if_name = sr_xpath_key_value(new_val ? new_val->xpath : old_val->xpath,
+                                     "interface", "name", &xpath_ctx);
+
+        intf = interface::find(if_name);
+        if (nullptr == intf) {
+            SRP_LOG_ERR_MSG("Interface does not exist");
+            return SR_ERR_INVAL_ARG;
+        }
+
+        switch (op) {
+            case SR_OP_CREATED:
+            case SR_OP_MODIFIED:
+                if (new_val->data.bool_val)
+                    intf->set(interface::admin_state_t::UP);
+                else
+                    intf->set(interface::admin_state_t::DOWN);
+                break;
+            case SR_OP_DELETED:
+                intf->set(interface::admin_state_t::DOWN);
+                break;
+            default:
+                break;
+        }
+        sr_xpath_recover(&xpath_ctx);
+
+        sr_free_val(old_val);
+        sr_free_val(new_val);
+
+        /* Commit the changes to VOM DB and VPP */
+        if ( OM::write("enable/"+intf->key(), *intf) != rc_t::OK ) {
+            SRP_LOG_ERR_MSG("Fail writing changes to VPP");
```

Comment:
Memory Leak: sr_free_change_iter(iter) and sr_free_val(old_val) sr_free_val(new_val) has not been called